WordPress/gutenberg
# Link Control: Add content and media filtering tabs

Closes: #63683

## What?
This update adds tab-based filtering to the link control interface, introducing "Content" and "Media" tabs to better organize link suggestions.

## Why?
The current link control interface displays all link suggestions in a single list, making it challenging for users to find specific types of content quickly. By introducing tabs to filter between content types and media, we can improve the user experience by:

- Making it easier to find specific types of content
- Creating a clearer mental model of content organization

## How?
- Added two new tabs to the link control interface:

   - "Content" tab (default) - displays all post types and taxonomies
   - "Media" tab - displays only media items

- Implemented tab-based filtering using the existing WordPress tab control component

## Testing Instructions
1. Open the Block Editor
2. Insert any block that supports links (e.g., Paragraph)
3. Click the link button from the block toolbar
4. Verify that you see two tabs: "Content" and "Media"

5. Test the Content tab:
   - Confirm it shows posts, pages, and other content types
   - Verify search functionality works within this context

6. Test the Media tab:
    - Confirm it only shows media items
    - Verify search functionality filters media items appropriately

## Screencast 

https://github.com/user-attachments/assets/ce40c13e-7599-41b5-bdcf-11747b842937
